python/lsst/pipe/tasks/processCcdWithFakes.py

```diff
@@ -393,51 +394,70 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
     def run(self, fakeCats, exposure, skyMap, wcs=None, photoCalib=None, exposureIdInfo=None,
             icSourceCat=None, sfdSourceCat=None, externalSkyWcsGlobalCatalog=None,
             externalSkyWcsTractCatalog=None, externalPhotoCalibGlobalCatalog=None,
-            externalPhotoCalibTractCatalog=None):
-        """Add fake sources to a calexp and then run detection, deblending and measurement.
+            externalPhotoCalibTractCatalog=None, idGenerator=None):
+        """Add fake sources to a calexp and then run detection, deblending and
+        measurement.
 
         Parameters
         ----------
         fakeCats : `list` of `lsst.daf.butler.DeferredDatasetHandle`
-                    Set of tract level fake catalogs that potentially cover
-                    this detectorVisit.
+            Set of tract level fake catalogs that potentially cover this
+            detectorVisit.
         exposure : `lsst.afw.image.exposure.exposure.ExposureF`
-                    The exposure to add the fake sources to
+            The exposure to add the fake sources to
         skyMap : `lsst.skymap.SkyMap`
             SkyMap defining the tracts and patches the fakes are stored over.
-        wcs : `lsst.afw.geom.SkyWcs`
-                    WCS to use to add fake sources
-        photoCalib : `lsst.afw.image.photoCalib.PhotoCalib`
-                    Photometric calibration to be used to calibrate the fake sources
-        exposureIdInfo : `lsst.obs.base.ExposureIdInfo`
-        icSourceCat : `lsst.afw.table.SourceCatalog`
-                    Default : None
-                    Catalog to take the information about which sources were used for calibration from.
-        sfdSourceCat : `lsst.afw.table.SourceCatalog`
-                    Default : None
-                    Catalog produced by singleFrameDriver, needed to copy some calibration flags from.
+        wcs : `lsst.afw.geom.SkyWcs`, optional
+            WCS to use to add fake sources
```

Comment:
Thanks for the formatting cleanups. Mind adding punctuation to these while you're at it?